segmentation/utils/data_utils.py

```diff
@@ -18,33 +18,24 @@ def validate_paths(paths):
         Raises errors if any directory is out of scope or non-existent
     """
 
-    cwd_parts = pathlib.Path.cwd().parts
-
+    # if given a single path, convert to list
     if not isinstance(paths, list):
         paths = [paths]
 
     for path in paths:
         if not os.path.exists(path):
-            for part in cwd_parts:
-                if part == 'segmentation':
-                    break
-                if part not in pathlib.Path(path).parts:
-                    if 'segmentation' in pathlib.Path(path).parts:
-                        raise ValueError(
-                            f'The path, {pathlib.Path(path).resolve()},'
-                            f'starts outside of Docker\'s scope.\n'
-                            f'Use relative paths which go no higher in scope '
-                            f'than the segmentation folder')
-                    else:
+            if str(path).startswith('data'):
+                for parent in pathlib.Path(path).parents:
+                    if not os.path.exists(parent):
                         raise ValueError(
-                            f'The path, {pathlib.Path(path).resolve()}, '
-                            f'exists outside of Docker\'s scope.\n'
-                            f'Move data, files, etc from {pathlib.Path(path).resolve()} '
-                            f'into the segmentation folder.\n'
-                            f'Reference moved data using relative paths, '
-                            f'which go no higher in scope than the segmentation folder')
-
-            raise ValueError(f"The folder path {pathlib.Path(path).resolve()}, does not exist...")
+                            f'The folder, {parent}, was not found...')
```

Comment:
Because you're using the pathlib.parents function to generate the sub_paths to check for existence, the paths are checked in descending order from longest to shortest. 

i.e., for the following path:  'data/sub_dir1/typo_dir/sub_dir3/file'
We would first check 'data/sub_dir1/typo_dir/sub_dir3'
then 'data/sub_dir1/typo_dir'
then 'data/sub_dir1'

However, even the first path won't exist, since it's nested under a fake subdir. What would be great is if we identified the specific directory that is causing the typo in the string. So we'll want to loop over the pathlib.parents in the opposite order, so we reach a failure only when the typo is introduced. Then, we can print out just that element of the path, along with the true sub_dir that it belongs to, in order to help guide debugging.

Reply:
It looks to me like this test would still fail in the newest version, is that not the case? We're looping through the parents, meaning any path containing a mis-typed file will trigger on the last sub_dir